src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt

```diff
@@ -220,10 +230,19 @@ class DtlsStack(
         val bufCopy = BufferPool.getBuffer(len).apply {
             System.arraycopy(data, off, this, 0, len)
         }
-        if (!incomingProtocolData.offer(ByteBuffer.wrap(bufCopy, 0, len))) {
-            logger.warn("DTLS stack queue full, dropping packet")
-            BufferPool.returnBuffer(bufCopy)
-            numPacketDropsQueueFull++
+        synchronized(lock) {
+            if (!running) {
+                BufferPool.returnBuffer(bufCopy)
+                return
+            }
+            if (incomingProtocolData.size >= QUEUE_SIZE) {
+                logger.warn("DTLS stack queue full, dropping packet")
+                BufferPool.returnBuffer(bufCopy)
+                numPacketDropsQueueFull++
+                Unit
```

Comment:
What's the "Unit" for?

Reply:
a `synchronized` block returns a value, and the `if` & `else` branches return different types (`Int` and `boolean`), so without the `Unit` intellij complains about a "implicit cast to any" for the result :/